src/pfe/file-watcher/server/src/projects/projectSpecifications.ts

```diff
@@ -797,6 +801,61 @@ const changeMavenProperties = async function (mavenProperties: any, operation: O
     return;
 };
 
+/**
+ * @function
+ * @description Reconfig the statusPingTimeout for a project.
+ *
+ * @param statusPingTimeout <Required | string> - Required value for reconfiguration of statusPingTimeout.
+ * @param operation <Required | Operation> - Operation object for the project.
+ *
+ * @returns Promise<any>
+ */
+const changeStatusPingTimeout = async function (statusPingTimeout: string, operation: Operation): Promise<any> {
+
+    const projectInfo: ProjectInfo = operation.projectInfo;
+    const projectID = projectInfo.projectID;
+    const projectHandler = await projectExtensions.getProjectHandler(projectInfo);
+    // make sure the value we saved in projectinfo is an integer
+    let pingTimeoutInt: number = parseInt(statusPingTimeout);
+
+    if (!pingTimeoutInt) {
```

Comment:
Think we should use `if(isNaN(pingTimeoutInt))` here instead. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

Reply:
updated